discussions-api/src/main/java/org/xwiki/contrib/discussions/DiscussionsActorServiceResolver.java

```diff
@@ -40,5 +40,5 @@
      * @param type the type
      * @return the actor service, {@link Optional#empty()} if the resolution failed
      */
-    Optional<DiscussionsActorService> get(String type);
+    DiscussionsActorService get(String type);
```

Comment:
(small detail) Why is the Optional removed? I agree the default implementation does not return the empty value, but could be the case for another implementation.

Reply:
So I improved a bit the javadoc there to specify that the contract of this resolver should always fallback on a default service if a specific implementation for the type does not exist. So it should never be the case to return null on another implementation.